otter_welcome_buddy/cogs/new_user_joins.py

```diff
@@ -1,36 +1,64 @@
+import discord
 from discord.ext import commands
+from discord.ext.commands import Bot
+from discord.ext.commands import Context
 
-from otter_welcome_buddy.formatters import debug, messages
+from otter_welcome_buddy.common.constants import OTTER_ROLE
+from otter_welcome_buddy.common.constants import WELCOME_MESSAGES
+from otter_welcome_buddy.formatters import debug
+from otter_welcome_buddy.formatters import messages
 
 
 class Greetings(commands.Cog):
     """When a user joins, sends reactionable message"""
 
-    def __init__(self, bot, messages_dependency, debug_dependency):
-        self.bot = bot
-        self.messages_formatter = messages_dependency
-        self.debug_formatter = debug_dependency
+    def __init__(
+        self,
+        bot: Bot,
+        messages_dependency: type[messages.Formatter],
+        debug_dependency: type[debug.Formatter],
+    ) -> None:
+        self.bot: Bot = bot
+        self.messages_formatter: type[messages.Formatter] = messages_dependency
+        self.debug_formatter: type[debug.Formatter] = debug_dependency
 
-    def _command_message(self):
+    def _command_message(self) -> str:
         return self.messages_formatter.welcome_message()
 
     @commands.Cog.listener()
-    async def on_ready(self):
+    async def on_ready(self) -> None:
         """Ready Event"""
         print(self.debug_formatter.bot_is_ready())
 
     @commands.Cog.listener()
-    async def on_member_join(self, member):
-        """Send Welcome message to new member"""
-        await member.send(self.messages_formatter.welcome_message())
+    async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) -> None:
+        """Event fired when a user react to the welcome message, giving the entry role to him"""
+        if payload.member is None:
+            print(f"Missing member to add role in {__name__}")
+            return
+        if not WELCOME_MESSAGES or str(payload.message_id) in WELCOME_MESSAGES:
+            try:
+                guild = next(guild for guild in self.bot.guilds if guild.id == payload.guild_id)
+                role = discord.utils.get(guild.roles, name=OTTER_ROLE)
```

Comment:
what kind of role? Prefer more explicit names

Reply:
The role base of the server, let's say the one to be considered as a member, should I specify it more here or in the README?

Reply:
Is more clear now? 
https://github.com/Proyecto-Nutria/otter-welcome-buddy/blob/9221567fd6beee546d3b433a1903478c2229df32/README.md#configuration

Reply:
I think in both places as well, so we can save time when reviewing the README or writing new code.

Reply:
I'll change the variable name but don't want to explain every time that I use OTTER_ROLE what's that